app/views/external_works/fetch.js.erb

```diff
@@ -2,14 +2,17 @@
   $j('#external_work_author').val("<%= @external_work.author %>").change();
   $j('#external_work_title').val("<%= @external_work.title %>").change();
   $j('#external_work_summary').val("<%= escape_javascript(@external_work.summary&.html_safe) %>").change();
-  $j('#external_work_summary').html("<%= escape_javascript(@external_work.summary&.html_safe) %>").change();
   $j('#fetched').val("<%= @external_work.id %>");
   $j('#external_work_rating_string').val("<%= @external_work.rating_string %>");
-  var categories = <%= @external_work.category_strings.to_json.html_safe %>;
-  $j('input[name="external_work[category_strings][]"]').each(function() {
-    $j(this).prop("checked", categories.includes(this.value));
+  var work_categories = <%= @external_work.category_strings.to_json.html_safe %>;
+  var unwanted_categories = <%= (Category.canonical.pluck(:name) - @external_work.category_strings).to_json.html_safe %>;
+  $j.each(work_categories, function(i, category) {
+    $j("input[value='" + category + "']").prop('checked', true);
   });
-  $j('#external_work_fandom_string').val("<%= @external_work.fandom_string %>");
-  $j('#external_work_relationship_string').val("<%= @external_work.relationship_string %>");
-  $j('#external_work_character_string').val("<%= @external_work.character_string %>");
+  $j.each(unwanted_categories, function(i, category) {
+    $j("input[value='" + category + "']").prop('checked', false);
+  });
```

Comment:
This is completely hideous, but it refuses to work when attempting to use a function to determine whether the `checked` property should be true or false. And I don't mean it refused to check or uncheck the box -- I mean it wouldn't fill in any of the other fields, either.

If someone else wants to try, please feel free!

Reply:
I think the problem is that the tests are not consistently waiting for the AJAX request (fetching external work info) to finish. The failures are not because of the script to check/uncheck the categories.

We now have a step to wait for AJAX requests, which should be included in both the step to use the bookmarklet and the step to pick a URL from the autocomplete:

https://github.com/otwcode/otwarchive/blob/5c96d176853af424d981d9e6ac208cc61f3a3001/features/step_definitions/generic_steps.rb#L49-L51

Reply:
I can add that and try again, but I'll be shocked if it works: my various unsuccessful attempts were consistently failing both in tests _and_ in the browser (I think the time it happened worked the other day was caching), whereas this version and the previous version (without the unchecking function) consistently passed.

Reply:
Yeah, no, same result.

Reply:
It looks like the `includes` function isn't available in older browsers (including PhantomJS, I guess?), but it might work to use `indexOf` or `$j.inArray` instead (and check for `-1`).

Reply:
asdfkjlsjg

The best part is I Googled the `includes` function, but didn't check the browser compatibility. 🤦 

I will give it another whirl now that I have this information -- thanks!